FL/Fl.H

```diff
@@ -689,24 +691,35 @@ int main() {
     This is a bitfield of what shift states were on and what mouse buttons
     were held down during the most recent event.
 
-    The legal event state bits are:
-
-    - FL_SHIFT
-    - FL_CAPS_LOCK
-    - FL_CTRL
-    - FL_ALT
-    - FL_NUM_LOCK
-    - FL_META
-    - FL_SCROLL_LOCK
-    - FL_BUTTON1
-    - FL_BUTTON2
-    - FL_BUTTON3
-    
-    X servers do not agree on shift states, and FL_NUM_LOCK, FL_META, and
-    FL_SCROLL_LOCK may not work. The values were selected to match the
-    XFree86 server on Linux. In addition there is a bug in the way X works
-    so that the shift state is not correctly reported until the first event
-    <I>after</I> the shift key is pressed or released.
+  \note FLTK platforms differ in what Fl::event_state() returns when it is called
+        while a modifier key or mouse button is being pressed or released.
+  - Under X11 and Wayland, Fl::event_state() indicates the state of the modifier keys and
+    mouse buttons just \b prior to the event. Thus, during the \c FL_KEYDOWN event generated
+    when pressing the shift key, for example, the \c FL_SHIFT bit of event_state() is 0 and
+    becomes 1 only at the next event which can be any other event, including e.g. \c FL_MOVE.
+  - Under other platforms the reported state of modifier keys or mouse buttons includes that
+    of the key or button being pressed or released.
+  - Fl::event_state() returns the same value under all platforms when it's called while a
+    non-modifier key (e.g. a letter or function key) is being pressed or released.
+  - X servers do not agree on shift states, and \c FL_NUM_LOCK, \c FL_META, and \c FL_SCROLL_LOCK
+    may not work.
+  - The values were selected to match the XFree86 server on Linux.
+  \note This inconsistency \b may be fixed (on X11 and Wayland) in a later release.
+  The legal event state bits are:
+    | Device   | State Bit      | Function                | Since |
+    |----------|----------------|-------------------------|-------|
+    | Keyboard | FL_SHIFT       | Shift                   |       |
+    | Keyboard | FL_CAPS_LOCK   | Caps Lock               |       |
+    | Keyboard | FL_CTRL        | Ctrl                    |       |
+    | Keyboard | FL_ALT         | Alt                     |       |
+    | Keyboard | FL_NUM_LOCK    | Num Lock                |       |
+    | Keyboard | FL_META        | Meta, e.g. "Windows"    |       |
+    | Keyboard | FL_SCROLL_LOCK | Scroll Lock             |       |
+    | Mouse    | FL_BUTTON1     | left button             |       |
+    | Mouse    | FL_BUTTON2     | middle button           |       |
+    | Mouse    | FL_BUTTON3     | right button            |       |
+    | Mouse    | FL_BUTTON4     | side button 1 (back)    | 1.4.0 |
+    | Mouse    | FL_BUTTON5     | side button 2 (forward) | 1.4.0 |
```

Comment:
1.3.9?

Reply:
The original (1.4.0) docs list this as "since 1.4.0" because 1.3.9 did not yet include it. Once it will be backported to 1.3.x this should read "`1.3.10`".

I'll work on this PR after 1.4.0 RC3 or the final release, whichever comes first. I'd like to let the 1.4 code settle so we can see if there are any issues. The next 1.3 release should definitely be the last - unless unforeseen severe issues arise.

Reply:
Sounds good, I'll update the table to say 1.3.10.

> I'd like to let the 1.4 code settle so we can see if there are any issues. 

Would you be able to say if the API will be stable, even if issues arise?